src/makeRobot.js

```diff
@@ -38,7 +38,87 @@
  * @return {Robot}
  */
 function makeRobot(name, wheels, version) {
-  // write code here
+  const robot = {
+
+    name,
+    version,
+    wheels,
+    coords: {
+      x: 0,
+      y: 0,
+    },
+
+    goRight(value) {
+      if (value <= 0) {
+        return this;
+      }
+
+      if (!value) {
+        this.coords.x++;
+      } else {
+        this.coords.x += value;
+      }
+
+      return this;
+    },
```

Comment:
no needs to use such a complex solution. You can just add a default value for your parameter. Apply for similar cases as well
```suggestion
    goRight(value = 1) {
      if (value > 0) {
        this.coords.x += value;
      }
      
      return this;
    },
```